src/pybind/mgr/cephadm/module.py

```diff
@@ -1582,7 +1582,7 @@ def describe_service(self, service_type: Optional[str] = None, service_name: Opt
                     spec = ServiceSpec(
                         unmanaged=True,
                         service_type=daemon_type_to_service(dd.daemon_type),
-                        service_id=dd.service_id(),
+                        service_id=dd.service_id() if n != 'osd.None' else 'unmanaged',
```

Comment:
this seems awkward; maybe better to put the special osd ``spec = ...`` in the osd block above, and only look up ``n`` in ``spec_store`` if spec is None? it's more code but i think it's more clear?

Reply:
@liewegas : you comment drives me to the right way to do this ... more easy than we thought :-) ... it was trivial ...

Reply:
will that works for existing osds that don't have the service_name in the unit.meta file?  we want to make sure those appear as ``osd.unmanaged`` too...

Reply:
seems to be solved with:
https://github.com/ceph/ceph/pull/40736